goshopify.go

```diff
@@ -244,21 +244,33 @@ func (c *Client) NewRequest(method, relPath string, body, options interface{}) (
 	return req, nil
 }
 
-// NewClient returns a new Shopify API client with an already authenticated shopname and
+// MustNewClient returns a new Shopify API client with an already authenticated shopname and
 // token. The shopName parameter is the shop's myshopify domain,
 // e.g. "theshop.myshopify.com", or simply "theshop"
 // a.NewClient(shopName, token, opts) is equivalent to NewClient(a, shopName, token, opts)
-func (app App) NewClient(shopName, token string, opts ...Option) *Client {
+func (app App) NewClient(shopName, token string, opts ...Option) (*Client, error) {
 	return NewClient(app, shopName, token, opts...)
 }
 
 // Returns a new Shopify API client with an already authenticated shopname and
 // token. The shopName parameter is the shop's myshopify domain,
 // e.g. "theshop.myshopify.com", or simply "theshop"
-func NewClient(app App, shopName, token string, opts ...Option) *Client {
+// panics if an error occurs
+func MustNewClient(app App, shopName, token string, opts ...Option) *Client {
```

Comment:
@I'm not sold that we want to expose a possible panic from this library even if it's heavily documented. Is there a strong rationale as to why someone wouldn't prefer to do this in their client code? I'd say lets just remove the panic altogether, but that's certainly an opinion.

Reply:
For instance the prometheus library uses MustRegister() which will panic if the call fails so this is not without precedence in the golang ecosystem.

I dislike the panic and also the return error. It's so unnecessary. All that is checked inside the NewClient() call is the url.Parse() which is silly. It just doesn't seem worth having the error returned and checked for that. 

I kept the panic as MustNewClient() so it's a drop-in replacement for the current NewClient() - I don't feel strongly about keeping it but for instance in my code I would just call NewClient() and ignore the error - I know that the domain is good cause I take whatever comes from shopify and if it's not then 🤷‍♀️   - the call will fail later.